File: backend/app/schemas/trade.py
import uuid
from datetime import datetime

from pydantic import BaseModel, Field

from backend.app.models.trade import TradeSide, TradeStatus
from backend.app.schemas.collection import CollectionItemRead


class TradeCreateRequest(BaseModel):
    receiver_user_id: uuid.UUID
    offered_item_ids: list[uuid.UUID] = Field(min_length=1)
    requested_item_ids: list[uuid.UUID] = Field(min_length=1)
    message: str | None = Field(default=None, max_length=2000)


class TradeOfferItemRead(BaseModel):
    id: uuid.UUID
    trade_offer_id: uuid.UUID
    side: TradeSide
    collection_item_id: uuid.UUID
    collection_item: CollectionItemRead | None = None

    model_config = {"from_attributes": True}


class TradeOfferRead(BaseModel):
    id: uuid.UUID
    sender_user_id: uuid.UUID
    receiver_user_id: uuid.UUID
    status: TradeStatus
    message: str | None = None
    created_at: datetime
    updated_at: datetime
    expires_at: datetime | None = None
    items: list[TradeOfferItemRead] = Field(default_factory=list)

    model_config = {"from_attributes": True}


class TradeListResponse(BaseModel):
    total: int
    items: list[TradeOfferRead]

File: frontend/src/app/models/trade.ts
import { CollectionItemRead } from './collection';

export enum TradeSide {
  SENDER = 'sender',
  RECEIVER = 'receiver',
}

export enum TradeStatus {
  PENDING = 'pending',
  ACCEPTED = 'accepted',
  REJECTED = 'rejected',
  CANCELLED = 'cancelled',
  EXPIRED = 'expired',
}

export interface TradeCreateRequest {
  receiver_user_id: string;
  offered_item_ids: string[];
  requested_item_ids: string[];
  message?: string | null;
}

export interface TradeOfferItemRead {
  id: string;
  trade_offer_id: string;
  side: TradeSide;
  collection_item_id: string;
  collection_item: CollectionItemRead | null;
}

export interface TradeOfferRead {
  id: string;
  sender_user_id: string;
  receiver_user_id: string;
  status: TradeStatus;
  message: string | null;
  created_at: string;
  updated_at: string;
  expires_at: string | null;
  items: TradeOfferItemRead[];
}

export interface TradeListResponse {
  total: number;
  items: TradeOfferRead[];
}
